tools/ruff_baselined.py

```
@@ -0,0 +1,350 @@
+        return _function_cache[key]
+    tree = _parse(rel)
+    if tree is None:
+        _function_cache[key] = None
+        return None
+    best: tuple[int, str] | None = None  # (start_line, name)
+    stack: list[tuple[ast.AST, list[str]]] = [(tree, [])]
+    while stack:
+        node, path_names = stack.pop()
+        for child in ast.iter_child_nodes(node):
+            if isinstance(child, ast.ClassDef):
+                stack.append((child, path_names + [child.name]))
```

Ruff (0.15.10)

[warning] Consider `[*path_names, child.name]` instead of concatenation

Replace with `[*path_names, child.name]`

(RUF005)